### ISSUE TITLE
Variable labeled unused if used in list comprehension in inner function

### ISSUE THREAD

**Issue by mcrumiller** [astral-sh/ty#3590]
### Summary

Probably related to #3442, but there is no "double comprehension" here.

```python
def outer() -> None:
    a = 1

    def inner() -> list[int]:
        return [a + x for x in [1, 2, 3]]
        # return [a + 1]  # not reported unused if used outside of a comprehension
```

VSCode reports that `a` is unused.

**Comment by carljm** [comment]
Thanks for the report! This is ultimately the same as #3442, despite the somewhat simpler repro, so I'll mark as duplicate (but I posted the code snippet there).
